examples/lighting-app/nrf5/main/AppTask.cpp

```diff
@@ -0,0 +1,462 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    Copyright (c) 2019 Google LLC.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "AppTask.h"
+#include "AppEvent.h"
+#include "DataModelHandler.h"
+#include "LEDWidget.h"
+#include "LightingManager.h"
+#include "Server.h"
+
+#include "app_button.h"
+#include "app_config.h"
+#include "app_timer.h"
+#include "boards.h"
+
+#include "nrf_log.h"
+
+#include "FreeRTOS.h"
+
+#include <platform/CHIPDeviceLayer.h>
+
+#define FACTORY_RESET_TRIGGER_TIMEOUT 3000
```

Comment:
nit: since this is C++, maybe consider for files:
 - constexpr instead of #defines
 - anonymous namespace to avoid 'static' everywhere.